app/Http/Controllers/SpecialOffersController.php

```diff
@@ -52,17 +53,44 @@ public function store(StoreSpecialOfferRequest $request)
 
         $games = $request->get('games');
 
+        $check = true;
+
         foreach ($games as $game) {
             $product = Product::FindOrFail($game);
-            $price = $product->prices()->where('special_offer_id', null)->where('user_id', null)->orderBy('date', 'DESC')->first();
-            $specialOffer->prices()->create(['amount' => $request->get('price_coef') * $price->amount, 'product_id' => $game]);
+            $price = $product->base_price;
+
+            if (($request->get('price_coef') != null && $price != $request->get('specialProductPrice')[$game])) {
+                $check = false;
+            }
+        }
+        if ($check == true){
+            $specialProductPrice = $request->get('specialProductPrice');
```

Comment:
šitą kintamąjį gal net galima apsirašyti ir prieš pirmąjį foreachą ir jį naudoti abiejuose cikluose